spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/InternalRowWrapper.java

```diff
@@ -71,8 +77,12 @@ public <T> void set(int pos, T value) {
     row.update(pos, value);
   }
 
-  private static BiFunction<InternalRow, Integer, ?> getter(DataType type) {
+  private static BiFunction<InternalRow, Integer, ?> getter(Type icebergType, DataType type) {
```

Comment:
nit: Do we rename the `type` to make it clear that it is `sql type` of spark?

Reply:
I'd probably just keep it `type` to cause less changes